include/ideep/operators/conv.hpp

```diff
@@ -1339,19 +1339,15 @@ struct convolution_forward
     }
     attr_t op_attr = attr;
     op_attr.set_scratchpad_mode(dnnl::scratchpad_mode::user);
-#ifdef __aarch64__
-    auto pd = get_primitive_desc</*with_bias=*/false>(
-        src_query, weights_desc, tensor::desc(), dst_query, strides, dilates_,
-        padding_l, padding_r, 0 /*weights_hashkey*/, is_channels_last, op_attr, aalgorithm, apkind);
 
-    // embed group info into weights_desc
-    return tensor::desc(pd.first.weights_desc(), groups);
-#else
     auto pd = get_primitive_desc</*with_bias=*/false>(
         src_query, weights_desc, tensor::desc(), dst_query, strides, dilates_,
         padding_l, padding_r, is_channels_last, op_attr, aalgorithm, apkind);
 
     // embed group info into weights_desc
+#ifdef __aarch64__
+    return tensor::desc(pd.first.weights_desc(), groups);
+#else
     return tensor::desc(pd.weights_desc(), groups);
 #endif
```

Comment:
Hi @jgong5 , thanks for the review. yes, we still need separate logic because of the need for caching the primitive on aarch64 vs primitive descriptor alone on x86. The fixed format kernels enabled us with handling the weights outside, but they are still mutable.  So, the state is maintained as the primitive resource, and caching the whole primitive is the only way to avoid redundant resource creation.

Reply:
Can you elaborate more on the difference of onednn primitives between aarch64 and x86. What resources need to be created?

Reply:
on x86, the kernels are initialized during the primitive init and will be cached as part of oneDNN primitive caching. Where as on aarch64, kernels are initialized during the resource mapper init, this is because the kernels are mutable. on aarch64, ACL kernel initialization involves workspace buffer allocations which are time consuming. This [PR](https://github.com/oneapi-src/oneDNN/pull/1455) has more details about why we can't use oneDNN primitive caching on aarch64 yet, and using ideep operator level caching is the only option.

Reply:
OK. Thanks for the detailed info. Do you have plan to make ACL immutable?